rails/app/javascript/components/Map.jsx

```diff
@@ -34,15 +34,64 @@ export default class Map extends Component {
   };
 
   componentDidMount() {
-    this.map = new mapboxgl.Map({
-        container: this.mapContainer,
-        style: this.props.mapboxStyle,
-        center: [this.props.centerLong, this.props.centerLat],
-        zoom: this.props.zoom,
-        maxBounds: this.checkBounds(), // check for bounding box presence
-        pitch: this.props.pitch,
-        bearing: this.props.bearing,
-        projection: this.props.mapProjection
+    if (this.props.useLocalMapServer) {
+      import('!maplibre-gl').then(module => {
+        this.initializeMap(module.default, true);
+      });
+    } else {
+      Promise.all([
+        import('!mapbox-gl'),
+        import('../vendor/mapboxgl-control-minimap.js')
+      ]).then(([mapboxGLModule, minimapModule]) => {
+        this.Minimap = minimapModule.default;
+        this.initializeMap(mapboxGLModule.default, false);
+      });
+    }
+  }
```

Comment:
I'm wondering if we could move the conditional imports into the constructor rather than on mount. We likely want the libraries imported and ready to go before the component has mounted and our rendering starts. I have an unsubstantiated concern that we may inadvertently re-import these libraries if React decides it needs to remount.

Reply:
So, I couldn't quite figure this out -- to the best of my knowledge and research, dynamic imports are inherently asynchronous, whereas constructors are not and can't handle Promises. Do you have any suggestions for how to handle this via the constructor?

However, I did implement a check to see if the modules have already been imported by setting that in the state. If they haven't, we import and store the modules, and if not, we skip and call `initializeMap` with the already imported modules.